### ISSUE TITLE
 FDS fails when trying to write to a READONLY file

### ISSUE THREAD

**Issue by gforney** [firemodels/fds#165]
```
Enclosed follows a case that behaves strange. With no sprinkler sentence 
activated there is a flash over after about 120 seconds. With the sentence 
&PART ID=’Water drops’....... activated, there is no fire at all (I ran 
this case for about 250 seconds). Any idea what the problem could be ?

Attached is the TINE.FDS which want burn – remove the &PART ID =’Water 
drops’.... and there will be a flash over

Regards 
Finn Drangsholt

```

Original issue reported on code.google.com by `Finn.Drangsholt` on 2007-09-25 06:17:15

<hr>
- _Attachment: [TINE.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-186/comment-0/TINE.fds)_

**Comment by gforney** [comment]
```
I'll take a look at it.
```

Original issue reported on code.google.com by `mcgratta` on 2007-09-25 12:21:22

**Comment by gforney** [comment]
```
I need the FDS version number. Are you running in parallel (MPI)? Do you see fire 
spread before flashover. I do not understand when you say "there is no fire at all".

Does this mean the fire disappears at 120 s? Or does this mean that the fire never

appears?
```

Original issue reported on code.google.com by `mcgratta` on 2007-09-25 12:47:13

**Comment by gforney** [comment]
```
I am using version 5_RC8 single CPU version. If you look at the attached file you 
may understand what I mean. 
```

Original issue reported on code.google.com by `Finn.Drangsholt` on 2007-09-25 14:14:01

<hr>
- _Attachment: [Tine-pic.pdf](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-186/comment-3/Tine-pic.pdf)_

**Comment by gforney** [comment]
```
I ran the case with and without the PART line. I used the latest version of FDS (SVN

Revision 677, Monday Sept 24). I ran the cases on a Linux machine. My results for 
HRR are attached. The PART line automatically adds water vapor to the air (40% 
relative humidity). Without it, the air is just 21% oxygen and 79% nitrogen (volume

fraction). So there is a slight difference in the composition of the air, leading to

a slight difference in radiation heat transfer and a slight difference in HRR. I did

not see the dramatically different results that you did, but it is very possible 
that something like this can happen. It appears that the IGNITOR readily ignites the

first "patch" of cardboard, and the HRR grows slightly as it burns. When it burns 
out, the HRR descreases and then starts to rise again as the next "patch" burns. I

am telling you what FDS is doing. In reality, the process is more continuous, but it

could happen that one experiment leads to flame spread and the other dies out. So I

do not know if this is a bug -- I think it is more a limitation of the model and a

fact of life with a coarse grid. All of the combustion is limited to a few grid 
cells. You have grid sensitivity issues to contend with in the beginning of the 
calculation. Once the fire starts spreading, the results are less sensitive to the

grid size.

Try downloading a more recent executable from the Downloads page and try again. 
Also, look at the HRR output like I have, at least in the beginning -- do you see 
dramatically different HRRs in the beginning? Or is that one case just has enough to

spread and the other does not? 

In the future, you need to decide if the objective of the calculation is to 
demonstrate that the fire will/will not spread; or if the objective is to predict 
the rate of growth and the HRR. If it is the former, you need a finer grid, at the

very least. If it is the latter, I suggest you use a larger ignition source so that

you minimize grid sensitivity issues.


```

Original issue reported on code.google.com by `mcgratta` on 2007-09-25 19:57:46

<hr>
- _Attachment: [TINE.PDF](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-186/comment-4/TINE.PDF)_

**Comment by gforney** [comment]
```
I increased the number of meshes from 2 to 3 in order to make finer grid around the

ignition source. However the simulation stops after a few iterations with forrtl :

sever <47> (se the attached file). I used the latest FDS available RC8_598. 

The sentence I used in TINE.FDS was :

&MESH IJK=15, 50, 50, XB=0.0, 3.0, 0.0,10.0, 0.0,10.0 / 
&MESH IJK=40,100,100, XB=3.0, 7.0, 0.0,10.0, 0.0,10.0 / 
&MESH IJK=15, 50, 50, XB=7.0,10.0, 0.0,10.0, 0.0,10.0 /

It seems that the problem occurs when I increase the number of meshes from 2 to 3

```

Original issue reported on code.google.com by `Finn.Drangsholt` on 2007-09-26 13:20:20

<hr>
- _Attachment: [Error.pdf](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-186/comment-5/Error.pdf)_

**Comment by gforney** [comment]
```
This latest bug has to do with output files on Windows machines. I will fix it.
```

Original issue reported on code.google.com by `mcgratta` on 2007-09-26 14:50:23
- **Labels added**: OpSys-Windows

**Comment by gforney** [comment]
```
Set FLUSH_FILE_BUFFERS=.FALSE. on the MISC line. This is a temporary fix to the last

problem that you reported. Let me know if that solves the problem. Note that 
Smokeview will not be fully functional until the calculation finishes because of 
this option.
```

Original issue reported on code.google.com by `mcgratta` on 2007-09-26 21:59:53

**Comment by gforney** [comment]
```
I tried to add FLUSH_FILE_BUFFERS=.FALSE. on the MISC line this way :
&MICH SURF_DEFAULT='CONCRET_CONSTRUCTION', FLUSH_FILE_BUFFERS=.FALSE. /
and I got the ERROR:Problem with MISC line
```

Original issue reported on code.google.com by `Finn.Drangsholt` on 2007-09-27 08:53:42

**Comment by gforney** [comment]
```
The input is on the &DUMP line not the &MISC line
```

Original issue reported on code.google.com by `drjfloyd` on 2007-09-27 12:39:43

**Comment by gforney** [comment]
```
I swapped to another computer.  I am now not able to reproduce the forrtl : 
sever <47>. The difference between these computers is the amount of RAM 1GB versus

2GB and processors ADM versus Intel

```

Original issue reported on code.google.com by `Finn.Drangsholt` on 2007-09-28 09:40:56

**Comment by gforney** [comment]
```
We cannot reproduce your problem except on Intel-based Windows machines. Linux and 
Macs do not have this problem. I debugged your case and found that opening the 
boundary (BNDF) file caused one of the slice (SLCF) files to close, even though the

file names and units are completely different. During a calculation, we close and 
then re-open the output files to force their contents to be written out to the disk,

so that you can run Smokeview during the FDS calculation. However, you and another

beta tester have reported this "READONLY" file problem. Each time, my only solution

to the problem was to renumber the output files (Fortran associates output file 
names with a "unit" number), and then the cases ran properly. I think it is a bug 
with the compiler or the OS, or both, but it is difficult to report because it is 
very hard to reproduce, and as you have seen, does not always occur as you go from

one machine to another. I am going to keep this Issue "On Hold" as a reminder that

the problem may not be entirely fixed.
```

Original issue reported on code.google.com by `mcgratta` on 2007-09-28 12:46:40
- **Labels added**: Usability

**Comment by gforney** [comment]
```
I am going to close this case to clean off my to-do list. I have not had any further

reports of this behavior. If it happens again, open up a new Issue Tracker, or add

to this one, which will be permanently archived.
```

Original issue reported on code.google.com by `mcgratta` on 2007-10-12 19:56:54

**Comment by gforney** [comment]
```
I have got similar problem when I try to run a model using FDS5 Version 5.1.0, Sun 
Revision No. 1162.

forrtl: severe (47): write to READONLY file, unit 2537

There was a similar problem reported by martino....@ticino.com, Sep 24, 2007 in 
issue tracker. It seems that there is a bug when I generate a data file using 
ProySim 2007. Could you kindly tell me how to sort out this problem?

http://code.google.com/p/fds-smv/issues/detail?id=179&can=1&q=readonly

I have attached the screen display and the FDS5 file for your reference.

```

Original issue reported on code.google.com by `david.chan2@fabermaunsell.com` on 2008-07-23 14:03:49

<hr>
- _Attachment: [Readonly Error.doc](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-186/comment-13/Readonly Error.doc)_
- _Attachment: [Floor10.fds](https://storage.googleapis.com/google-code-attachments/fds-smv/issue-186/comment-13/Floor10.fds)_

**Comment by gforney** [comment]
```
Could you try this case using the latest version of FDS. Also, does the problem 
persist if you use FLUSH_FILE_BUFFERS=.FALSE. on the MISC line?
```

Original issue reported on code.google.com by `mcgratta` on 2008-07-23 14:10:31

**Comment by gforney** [comment]
```
Correction -- FLUSH_FILE_BUFFERS=.FALSE. goes on the DUMP line.
```

Original issue reported on code.google.com by `mcgratta` on 2008-07-23 14:12:12

**Comment by gforney** [comment]
```
Has your problem been resolved?
```

Original issue reported on code.google.com by `mcgratta` on 2008-08-08 19:36:24

**Comment by gforney** [comment]
```
Yes, I think this issue should be closed
```

Original issue reported on code.google.com by `Finn.Drangsholt` on 2008-08-09 21:52:35

**Comment by gforney** [comment]
```
(No text was entered with this change)
```

Original issue reported on code.google.com by `mcgratta` on 2008-09-09 15:00:00
